Index: evolution-data-server-2.32.2/addressbook/backends/google/e-book-backend-google.c
===================================================================
--- evolution-data-server-2.32.2.orig/addressbook/backends/google/e-book-backend-google.c	2010-11-22 22:26:37.000000000 -0500
+++ evolution-data-server-2.32.2/addressbook/backends/google/e-book-backend-google.c	2011-04-06 16:25:52.028825647 -0400
@@ -511,7 +511,7 @@
 		GTimeVal updated;
 
 		g_assert (g_time_val_from_iso8601 (last_updated, &updated) == TRUE);
-		gdata_query_set_updated_min (query, &updated);
+		gdata_query_set_updated_min (query, updated.tv_sec);
 		gdata_contacts_query_set_show_deleted (GDATA_CONTACTS_QUERY (query), TRUE);
 	}
 
@@ -789,9 +789,9 @@
 
 	/* Update the contact on the server */
 	new_entry = GDATA_ENTRY (
-		gdata_contacts_service_update_contact (
-			GDATA_CONTACTS_SERVICE (priv->service),
-			GDATA_CONTACTS_CONTACT (entry),
+		gdata_service_update_entry (
+			priv->service,
+			entry,
 			NULL, &error));
 	g_object_unref (entry);
 
Index: evolution-data-server-2.32.2/configure.ac
===================================================================
--- evolution-data-server-2.32.2.orig/configure.ac	2010-12-27 08:19:33.000000000 -0500
+++ evolution-data-server-2.32.2/configure.ac	2011-04-06 16:25:19.628825650 -0400
@@ -325,12 +325,6 @@
 	libsoup-2.4 >= libsoup_minimum_version
 	libgdata >= libgdata_minimum_version])
 
-dnl libgdata 0.7 will break API.  We're not going to deal with
-dnl that on a stable branch, so cap the supported version.
-if $PKG_CONFIG --atleast-version=0.7 libgdata; then
-	AC_MSG_ERROR([libgdata 0.7 is not supported, use 0.6 instead])
-fi
-
 LIBICAL_REQUIRED=libical_minimum_version
 AC_SUBST(LIBICAL_REQUIRED)
 
