From e7201c8163317afce06c973e38df829e68869acd Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 12 Jan 2011 15:36:56 +0000
Subject: Bug #639165 - maildir expunge not working as expected

---
diff --git a/camel/providers/local/camel-local-provider.c b/camel/providers/local/camel-local-provider.c
index a136703..bbaf97c 100644
--- a/camel/providers/local/camel-local-provider.c
+++ b/camel/providers/local/camel-local-provider.c
@@ -164,7 +164,7 @@ local_url_hash (gconstpointer v)
 	const CamelURL *u = v;
 	guint hash = 0;
 
-#define ADD_HASH(s) if (s) hash ^= g_str_hash (s);
+#define ADD_HASH(s) if (s && *s) hash ^= g_str_hash (s);
 
 	ADD_HASH (u->protocol);
 	ADD_HASH (u->user);
--
cgit v0.8.3.1
