From 4913827c8936a378b0adbbdab8b2ad609dc0f35d Mon Sep 17 00:00:00 2001
From: Andrew McMillan <andrew@morphoss.com>
Date: Thu, 13 Jan 2011 21:58:46 +0000
Subject: Bug #639469 - WebDAV addressbook only uses https

---
Index: evolution-data-server-2.32.1/addressbook/backends/webdav/e-book-backend-webdav.c
===================================================================
--- evolution-data-server-2.32.1.orig/addressbook/backends/webdav/e-book-backend-webdav.c	2011-01-19 08:15:47.247655260 -0500
+++ evolution-data-server-2.32.1/addressbook/backends/webdav/e-book-backend-webdav.c	2011-01-19 08:16:15.446914992 -0500
@@ -1234,7 +1234,7 @@
 	if (use_ssl != NULL && strcmp (use_ssl, "1") == 0) {
 		soup_uri_set_scheme (suri, "https");
 	} else {
-		soup_uri_set_scheme (suri, "https");
+		soup_uri_set_scheme (suri, "http");
 	}
 
 	/* append slash if missing */
