From 9b014401e787fdd2d95abdf23a8568408ea96d17 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Tue, 11 Jan 2011 13:35:55 +0000
Subject: Bug #525565 - Crash in find_book_by_contact at e-name-selector-entry.c

---
diff --git a/libedataserverui/e-name-selector-entry.c b/libedataserverui/e-name-selector-entry.c
index e135847..0340fba 100644
--- a/libedataserverui/e-name-selector-entry.c
+++ b/libedataserverui/e-name-selector-entry.c
@@ -2299,7 +2299,7 @@ find_book_by_contact (GList *books, const gchar *contact_uid)
 
 	for (l = books; l; l = g_list_next (l)) {
 		EBook    *book = l->data;
-		EContact *contact;
+		EContact *contact = NULL;
 		gboolean  result;
 
 		result = e_book_get_contact (book, contact_uid, &contact, NULL);
@@ -2349,7 +2349,6 @@ editor_closed_cb (GtkWidget *editor, gpointer data)
 
 	g_free (contact_uid);
 	g_object_unref (contact);
-	g_object_unref (editor);
 	g_object_unref (name_selector_entry);
 }
 
--
cgit v0.8.3.1
