--- evolution-data-server-2.32.3/libedataserver/e-proxy.c	2011-04-21 21:35:37.000000000 +0200
+++ evolution-data-server-2.32.3/libedataserver/e-proxy.c	2014-06-13 09:27:41.256199187 +0200
@@ -30,6 +30,7 @@
 
 #ifdef _WIN32
 #include <winsock2.h>
+#include <ws2tcpip.h>
 #ifndef IN6_ARE_ADDR_EQUAL
 #define IN6_ARE_ADDR_EQUAL(a, b)	\
     (memcmp ((gpointer)(a), (gpointer)(b), sizeof (struct in6_addr)) == 0)
