--- evolution-data-server-2.32.3/addressbook/libedata-book/e-data-book.c	2011-04-21 21:35:36.000000000 +0200
+++ evolution-data-server-2.32.3/addressbook/libedata-book/e-data-book.c	2011-09-30 09:24:34.941032359 +0200
@@ -632,7 +632,7 @@
 
 	return g_strdup_printf ("/org/gnome/evolution/dataserver/addressbook/BookView/%d/%d",
 				getpid (),
-				g_atomic_int_exchange_and_add ((int*)&counter, 1));
+				g_atomic_int_add ((int*)&counter, 1));
 }
 
 static gboolean
--- evolution-data-server-2.32.3/addressbook/libedata-book/e-data-book-factory.c	2011-04-21 21:36:35.000000000 +0200
+++ evolution-data-server-2.32.3/addressbook/libedata-book/e-data-book-factory.c	2011-09-30 11:35:59.042686099 +0200
@@ -229,7 +229,7 @@
 
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/addressbook/%d/%u",
-		getpid (), g_atomic_int_exchange_and_add (&counter, 1));
+		getpid (), g_atomic_int_add (&counter, 1));
 }
 
 static gboolean
--- evolution-data-server-2.32.3/calendar/libedata-cal/e-data-cal-factory.c	2011-04-21 21:36:35.000000000 +0200
+++ evolution-data-server-2.32.3/calendar/libedata-cal/e-data-cal-factory.c	2011-09-30 11:51:31.091013649 +0200
@@ -245,7 +245,7 @@
 
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/calendar/%d/%u",
-		getpid (), g_atomic_int_exchange_and_add (&counter, 1));
+		getpid (), g_atomic_int_add (&counter, 1));
 }
 
 static gboolean
--- evolution-data-server-2.32.3/libedataserver/e-data-server-util.c	2011-04-21 21:35:37.000000000 +0200
+++ evolution-data-server-2.32.3/libedataserver/e-data-server-util.c	2011-09-30 09:18:51.744293058 +0200
@@ -245,7 +245,7 @@
 static gunichar
 stripped_char (gunichar ch)
 {
-	gunichar *decomp, retval;
+	gunichar decomp[4], retval;
 	GUnicodeType utype;
 	gsize dlen;
 
@@ -255,16 +255,15 @@
 	case G_UNICODE_CONTROL:
 	case G_UNICODE_FORMAT:
 	case G_UNICODE_UNASSIGNED:
-	case G_UNICODE_COMBINING_MARK:
+	case G_UNICODE_SPACING_MARK:
 		/* Ignore those */
 		return 0;
 	default:
 		/* Convert to lowercase, fall through */
 		ch = g_unichar_tolower (ch);
 	case G_UNICODE_LOWERCASE_LETTER:
-		if ((decomp = g_unicode_canonical_decomposition (ch, &dlen))) {
+		if ((dlen = g_unichar_fully_decompose (ch, FALSE, decomp, 4))) {
 			retval = decomp[0];
-			g_free (decomp);
 			return retval;
 		}
 		break;
