--- evolution-exchange-2.32.3/eplugin/exchange-operations.c	2011-04-21 21:35:56.000000000 +0200
+++ evolution-exchange-2.32.3/eplugin/exchange-operations.c	2012-04-24 10:18:57.649271541 +0200
@@ -63,7 +63,7 @@
 {
 	if (!exchange_global_config_listener) {
 		exchange_global_config_listener = exchange_config_listener_new ();
-		g_atexit (free_exchange_listener);
+		atexit (free_exchange_listener);
 	}
 	return 0;
 }
--- evolution-exchange-2.32.3/server/lib/e2k-context.h	2011-04-21 21:35:57.000000000 +0200
+++ evolution-exchange-2.32.3/server/lib/e2k-context.h	2012-05-08 12:33:04.206153119 +0200
@@ -4,6 +4,10 @@
 #ifndef __E2K_CONTEXT_H__
 #define __E2K_CONTEXT_H__
 
+#ifdef DATADIR
+#undef DATADIR
+#endif
+
 #include <libsoup/soup-message.h>
 #include <libsoup/soup-session.h>
 #include <sys/time.h>
--- evolution-exchange-2.32.3/tools/ximian-connector-setup.c	2011-04-21 21:35:58.000000000 +0200
+++ evolution-exchange-2.32.3/tools/ximian-connector-setup.c	2012-04-24 10:20:22.351154684 +0200
@@ -89,7 +89,6 @@
 	textdomain (GETTEXT_PACKAGE);
 
 	g_type_init ();
-	g_thread_init (NULL);
 	gtk_init_with_args (&argc, &argv, NULL, NULL, (gchar *)GETTEXT_PACKAGE, &error);
 
 	if (error != NULL) {
