--- evolution-2.32.3/calendar/gui/alarm-notify/notify-main.c	2011-03-29 20:38:46.000000000 +0200
+++ evolution-2.32.3/calendar/gui/alarm-notify/notify-main.c	2012-04-04 11:29:12.805728973 +0200
@@ -87,8 +87,6 @@
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	g_thread_init (NULL);
-
 #ifdef G_OS_WIN32
 	path = g_build_path (";", _e_get_bindir (), g_getenv ("PATH"), NULL);
 
--- evolution-2.32.3/capplet/anjal-settings-main.c	2011-03-29 20:38:46.000000000 +0200
+++ evolution-2.32.3/capplet/anjal-settings-main.c	2012-04-04 11:29:05.405914580 +0200
@@ -247,9 +247,6 @@
 		g_error_free(error);
 	}
 
-	if (!g_thread_get_initialized ())
-		g_thread_init (NULL);
-
 	client = gconf_client_get_default();
 
 	e_passwords_init();
--- evolution-2.32.3/capplet/settings/mail-guess-servers.c	2011-03-29 20:38:46.000000000 +0200
+++ evolution-2.32.3/capplet/settings/mail-guess-servers.c	2012-04-04 11:28:56.432139659 +0200
@@ -325,7 +325,6 @@
 main (gint argc, gchar **argv)
 {
 	EmailProvider *provider;
-	g_thread_init (NULL);
 	g_type_init ();
 
 	provider = g_new0(EmailProvider, 1);
--- evolution-2.32.3/plugins/backup-restore/backup.c	2011-03-29 20:38:48.000000000 +0200
+++ evolution-2.32.3/plugins/backup-restore/backup.c	2012-04-04 11:29:40.086044709 +0200
@@ -663,8 +663,6 @@
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	g_thread_init (NULL);
-
 	gtk_init_with_args (
 		&argc, &argv, NULL, options, (gchar *) GETTEXT_PACKAGE, &error);
 	if (error != NULL) {
--- evolution-2.32.3/shell/main.c	2011-03-29 20:38:57.000000000 +0200
+++ evolution-2.32.3/shell/main.c	2012-04-04 11:29:20.453537149 +0200
@@ -520,8 +520,6 @@
 	}
 
 	g_type_init ();
-	if (!g_thread_get_initialized ())
-		g_thread_init (NULL);
 
 	#ifdef HAVE_ICAL_UNKNOWN_TOKEN_HANDLING
 	ical_set_unknown_token_handling_setting (ICAL_DISCARD_TOKEN);
