--- modules/mailto-handler/apps-evolution-mail-prompts-checkdefault.schemas.in	(revision 37459)
+++ modules/mailto-handler/apps-evolution-mail-prompts-checkdefault.schemas.in	(working copy)
@@ -5,7 +5,7 @@
       <applyto>/apps/evolution/mail/prompts/checkdefault</applyto>
       <owner>evolution-mail</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
         <short>Check whether Evolution is the default mailer</short>
         <long>Every time Evolution starts, check whether or not it is the default mailer.</long>
