--- evolution-2.32.3/plugins/pst-import/pst-importer.c	2011-03-29 20:38:48.000000000 +0200
+++ evolution-2.32.3/plugins/pst-import/pst-importer.c	2014-02-26 11:39:35.755184800 +0100
@@ -1690,7 +1690,7 @@
 	DEBUG_REGISTER_CLOSE ();
 #endif
 
-	if (pst_open (pst, filename) < 0) {
+	if (pst_open (pst, filename, "utf-8") < 0) {
 		pst_error_msg ("Error opening PST file %s", filename);
 		return -1;
 	}
