--- evolution-2.32.3/configure.ac	2011-03-29 20:38:47.000000000 +0200
+++ evolution-2.32.3/configure.ac	2012-04-24 09:29:54.230635174 +0200
@@ -303,6 +303,7 @@
 dnl **********************************
 PKG_CHECK_MODULES([GNOME_PLATFORM],
 	[gio-2.0 >= glib_minimum_version
+	 gmodule-2.0 >= glib_minimum_version
 	 $GTK >= $gtk_minimum_version
 	 $GAIL >= $gail_minimum_version
 	 gconf-2.0 >= gconf_minimum_version
