--- evolution-2.32.3/configure.ac	2011-03-29 20:38:47.000000000 +0200
+++ evolution-2.32.3/configure.ac	2012-08-29 09:03:48.215645053 +0200
@@ -23,7 +23,7 @@
 dnl Autoconf / Automake Initialization
 AC_PREREQ(2.58)
 AC_INIT([evolution],[evo_version],[http://bugzilla.gnome.org/enter_bug.cgi?product=Evolution])
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror foreign])
+AM_INIT_AUTOMAKE([1.10 -Wall foreign])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_SRCDIR(README)
@@ -202,6 +202,7 @@
 AC_SUBST([UPGRADE_REVISION],[upgrade_revision])
 AC_DEFINE_UNQUOTED(UPGRADE_REVISION, ["$UPGRADE_REVISION"], [The number of times we've upgraded since the BASE_VERSION release])
 
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 AC_PROG_CC
 AC_C_INLINE
 AC_PROG_CPP
